src/OneDriveSdk.WinRT/Authentication/AdalServiceInfoProvider.cs

```diff
@@ -65,12 +64,14 @@ public AdalServiceInfoProvider(IAuthenticationProvider authenticationProvider, I
 
             var serviceInfo = await base.GetServiceInfo(appConfig, null, httpProvider, clientType);
 
-            serviceInfo.BaseUrl = appConfig.ActiveDirectoryServiceEndpointUrl;
             serviceInfo.ServiceResource = appConfig.ActiveDirectoryServiceResource;
 
-            if (string.IsNullOrEmpty(serviceInfo.ReturnUrl))
+            if (string.IsNullOrEmpty(serviceInfo.BaseUrl) && !string.IsNullOrEmpty(serviceInfo.ServiceResource))
             {
-                serviceInfo.ReturnUrl = WebAuthenticationBroker.GetCurrentApplicationCallbackUri().ToString();
+                serviceInfo.BaseUrl = string.Format(
+                    Constants.Authentication.OneDriveBusinessBaseUrlFormatString,
+                    serviceInfo.ServiceResource.TrimEnd('/'),
```

Comment:
Would it be possible to error out if the service resource has the trailing slash to force the proper value to be stored, if value could be used elsewhere.  I have wasted so much time thinking my value was 'correct'